s3_sync/s3_sync.py

```diff
@@ -0,0 +1,179 @@
+args = parser.parse_args()
+settings_file = args.settings_file
+
+# Read settings_file
+config = ConfigParser(interpolation = ExtendedInterpolation())
+config.read(settings_file)
+
+# Configure logbook logging
+logger = RotatingFileHandler(config['Logs']['logfile'],
+                             max_size = int(config['Logs']['max_size']),
+                             backup_count = int(config['Logs']['backup_count']),
+                             level = int(config['Logs']['level']))
+logger.push_application()
+logger = Logger(__name__)
+
+# Get Computer name
+computer_name = os.environ['COMPUTERNAME']
+
+
+def verify_settings_file_exists(settings_file):
+    """
+    Verify that settings file exists.
+
+    Args:
+      settings_file (str): Path for the local settings file.
+
+    Returns:
+      If file exists, return None, otherwise exit.
+    """
+    if os.path.exists(settings_file):
+        return
+    else:
+        logger.error("Settings file not found:", settings_file)
+        sys.exit("[-] Settings file not found")
+
+
+def set_environment_variables():
+    """
+    Set some of the read settings as environment variables.
+    """
+    os.environ['AWS_ACCESS_KEY_ID'] = config['AWS']['AWS_ACCESS_KEY_ID']
+    os.environ['AWS_SECRET_ACCESS_KEY'] = config['AWS']['AWS_SECRET_ACCESS_KEY']
+    os.environ['slack_webhook_url'] = config['Slack']['webhook_url']
+
+
+def verify_local_folder_exists(local_video_records_done_folder):
+    """
+    Check whether the local video records done folder exists
+
+    Args:
+      local_video_records_done_folder (str): The path of the local video
+        records done folder.
+
+    Returns:
+      If folder exists return None, and if not, logs error and exit.
+    """
+    if os.path.exists(local_video_records_done_folder):
+        return
+    else:
+        logger.error("Local Video Records Done folder not found")
+        sys.exit("[-] Local Video Records Done folder not found")
+
+
+def verify_aws_cli_installed(aws_cli_binary):
+    """
+    Check whether AWS CLI is installed
+
+    Args:
+      aws cli binary (str): absolute path to aws cli binary file.
+
+    Returns:
+      If file exists, return None, else log error and exit.
+
+    """
+    if os.path.exists(aws_cli_binary):
+        return
+    else:
+        logger.error("Could not find AWS CLI executable")
+        sys.exit("[-] Could not find AWS CLI executable")
+
+
+def verify_s3_bucket_exists(s3_bucket_name):
+    """
+    Check whether S3 bucket exists
+
+    Args:
+      s3_bucket_name (str): The s3 bucket name
+
+    Returns:
+      list: if connection established and bucket found, return list of
+        objects in bucket otherwise error and exit on any issues trying
+        to list objects in bucket.
+    """
+    try:
+        ls_s3_bucket_cmd = 'aws s3 ls {}'.format(s3_bucket_name)
+        subprocess.run(ls_s3_bucket_cmd, check=True,
+                       stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
Ideally we would use boto, but at this point it does not seem to have that functionality:
https://github.com/boto/boto3/issues/358